samples/client.c

```diff
@@ -437,21 +437,25 @@ access_region(int sock, int region, bool is_write, uint64_t offset,
             .iov_len = data_len
         }
     };
-    struct {
-        struct vfio_user_region_access region_access;
-        char data[data_len];
-    } __attribute__((packed)) recv_data;
-    int op, ret;
+
+    struct vfio_user_region_access *recv_data;
     size_t nr_send_iovecs, recv_data_len;
+    int op, ret;
 
     if (is_write) {
         op = VFIO_USER_REGION_WRITE;
         nr_send_iovecs = 3;
-        recv_data_len = sizeof(recv_data.region_access);
+        recv_data_len = sizeof(*recv_data);
     } else {
         op = VFIO_USER_REGION_READ;
         nr_send_iovecs = 2;
-        recv_data_len = sizeof(recv_data);
+        recv_data_len = sizeof (*recv_data) + data_len;
+    }
+
+    recv_data = calloc(1, recv_data_len);
```

Comment:
I think we can get away with using `alloca` here since this is a sample, we avoid the `free` later as well.

Reply:
I'm not a big fan of alloca(). I know it doesn't matter for the samples as you say, but since I've already written it, I think I'll leave it :)